src/ristretto/pedersen/extended_commitment_factory.rs

```diff
@@ -166,8 +165,9 @@ impl ExtendedHomomorphicCommitmentFactory for ExtendedPedersenCommitmentFactory
         k_vec: &[RistrettoSecretKey],
         v: &RistrettoSecretKey,
     ) -> Result<PedersenCommitment, CommitmentError> {
-        let blinding_factors: Vec<Scalar> = k_vec.iter().map(|k| k.0).collect();
+        let mut blinding_factors: Vec<Scalar> = k_vec.iter().map(|k| k.0).collect();
```

Comment:
Can you use a `Zeroizing` wrapper here instead? Then it'll drop if the `commit_scalars` call fails.

Reply:
good point, let me see

Reply:
It works! See [this branch](https://github.com/AaronFeickert/tari-crypto/commit/0a7e629c4cf4fa84af0957febbd9a62c1b4741f5) for an implementation.